plugins/woocommerce/includes/class-wc-order-factory.php

```diff
@@ -128,17 +133,16 @@ public static function get_orders( $order_ids = array(), $skip_invalid = false )
 			}
 		}
 
-		// restore the sort order.
-		$result = array_values( array_replace( array_flip( $order_ids ), $result ) );
-
 		if ( $use_orders_cache ) {
-			foreach ( $result as $order ) {
-				$order_cache->set( $order );
+			foreach ( $result as $order_id => $order ) {
+				$order_cache->set( $order, $order->get_id( $order_id ) );
```

Comment:
Unless I'm reading this wrong, I believe `$order->get_id()` doesn't admit params. Also, is there any benefit to passing the ID when the order cache would eventually call that same function to populate the ID?

Reply:
Ah that `$order_id` in param is a typo, fixed in https://github.com/woocommerce/woocommerce/pull/37192/commits/52890fccab8fa1f0e3539d25158073b64e3b75aa. 

Regarding passing the param to cache call, it's not needed, but given that we already have it available, I am hoping to prevent unnecessary calls. It doesn't matter much, but since we already have, why not.